contracts/base/RegistryManager.sol

```diff
@@ -29,8 +25,22 @@ contract RegistryManager is Ownable2Step {
         registry = _registry;
     }
 
+    /**
+     * @notice Checks if given integration address is listed and not flagged in the registry.
+     *         Reverts if given address is not-listed or flagged.
+     * @param integration Address of the integration
+     */
+    function checkPermittedIntegration(address integration) internal view {
+        // Only allow registered and non-flagged integrations
+        (uint64 listedAt, uint64 flaggedAt) = ISafeProtocolRegistry(registry).check(integration);
+        if (listedAt == 0 || flaggedAt != 0) {
+            revert IntegrationNotPermitted(integration, listedAt, flaggedAt);
+        }
+    }
+
     /**
      * @notice Allows only owner to update the address of a registry. Emits event RegistryChanged(egistry, newRegistry)
+     * @dev Do not set owner as a Safe address with Manager enabled as fallback handler.
```

Comment:
Could we add a "why" here

Reply:
Updated and Moved it to constructor: https://github.com/5afe/safe-core-protocol/pull/70/commits/7acc9cc13058551a41eb78f877f78fc35c9e9a6f